pytorch_lightning/accelerators/tpu_accelerator.py

```diff
@@ -261,10 +264,31 @@ def optimizer_step(self, optimizer, batch_idx, opt_idx, lambda_closure):
             using_lbfgs=is_lbfgs
         )
 
-    def clip_gradients(self, optimizer, clip_val=None):
-        # apply clip gradients
-        # TODO: separate TPU case from here
-        self._clip_gradients(optimizer, clip_val)
+    def _clip_gradients(self, optimizer, grad_clip_val):
```

Comment:
Can we have only one which takes `EPSILON`, either `EPSILON` or `FP16_EPSILON `. Less duplicated code.

Reply:
Maybe makes gradient_clipping_epsilon a property.

Reply:
Moved to properties contained in the respective classes

Reply:
The functions are different, the one in Apex is true to the torch code and the tpu one is based on the TPU guide suggestions, combining them is tricky and I think the logic should be kept separate...

Reply:
We have the epsilons defined already